adl_parser.py

```diff
@@ -0,0 +1,95 @@
+block_names = ("object", "region", "info", "table")
+table_k_v = "# val    err-     err+     xmin     xmax"
+
+
+def adl_parser(adl_file='adl.adl'):
+    """Returns the ADL parsed file as JSON object
+
+    :param type: File
+    :param adl_file: Input .adl from the user.
+    :return : JSON object
+    """
+    json_parsed = {
+        "object": [],
+        "region": [],
+        "info": [],
+        "table": []
+    }
+    block_data = {}
+    parsing_block = False
+    parsing_table = False
+
+    # Read the ADL file
+    f = open(adl_file, 'r')
+
+    for line in f:
```

Comment:
Steps to take:

(1) remove all comment lines (maybe we might need the ones that act as headers to `table` blocks)
(2) Search for blocks or `define` 
   - [ ] when block: (1) get whole block content (2) parse block content 
   - [ ] when define: (1) add variable to definition array
 
(3) Parse block content:
   - [ ] check for keywords

keyword name | functionality
-- | --
define | Define variables, constants.
select | Select objects or events based on criteria that follow the keyword.
reject | Reject objects or events based on criteria that follow the keyword.
take / using / : | Define the mother object type in an object block.
sort | sorts an object in an ascending or descending order with respect to a property.
weight | Weight the events with a (currently constant) number in the region block.
histo | Fill histograms.

  - [ ] in `region` blocks sometime the first line show inheritance from other region (so if (1)first lines of the block (2) not from above keywords (3) names of previous `regions` added, we should keep in an `parent` array. e.g.
```# Many bins njets, nb, LT, dphilW, HT
region multib
  baseline
  select size(bjets) > 0
  select size(jets) >= 6

# aggregate zerob search regions
region multibSR1
  multib
  select size(bjets) >= 1
  select size(jets) >= 6
  select LepT >= 600
  select dphilW > 0.5
  select HT >= 1000
```
  - [ ] when there is an error parsing a block don't return that block, move to next
  - [ ] split code in smaller method for parsing each block, etc